biz.aQute.repository/src/aQute/bnd/deployer/repository/ProgressWrappingStream.java

```diff
@@ -12,8 +15,60 @@
 	private Task		task;
 
 	public ProgressWrappingStream(InputStream delegate, String name, int size, ProgressPlugin progressPlugin) {
+		this(delegate, name, size, Collections.singletonList(progressPlugin));
```

Comment:
What if the arg `progressPlugin` is null? This method creates a single element list with null which will blow up in the other constructor which assumes the list holds real values.